builtin/providers/google/resource_compute_disk.go

```diff
@@ -171,16 +175,54 @@ func resourceComputeDiskRead(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	disk, err := config.clientCompute.Disks.Get(
-		project, d.Get("zone").(string), d.Id()).Do()
+	region, err := getRegion(d, config)
 	if err != nil {
-		return handleNotFoundError(err, d, fmt.Sprintf("Disk %q", d.Get("name").(string)))
+		return err
 	}
 
+	getDisk := func(zone string) (interface{}, error) {
+		return config.clientCompute.Disks.Get(project, zone, d.Id()).Do()
+	}
+
+	var disk *compute.Disk
+	var e error
+	if zone, ok := d.GetOk("zone"); ok {
+		disk, err = config.clientCompute.Disks.Get(
+			project, zone.(string), d.Id()).Do()
+		if err != nil {
+			return handleNotFoundError(err, d, fmt.Sprintf("Disk %q", d.Get("name").(string)))
+		}
+	} else {
+		// If the resource was imported, the only info we have is the ID. Try to find the resource
+		// by searching in the region of the project.
+		var resource interface{}
+		resource, e = getZonalResourceFromRegion(getDisk, region, config.clientCompute, project)
+
+		if e != nil {
+			return e
+		}
+
+		disk = resource.(*compute.Disk)
+	}
+
+	zoneUrl := strings.Split(disk.Zone, "/")
+	typeUrl := strings.Split(disk.Type, "/")
+	d.Set("name", disk.Name)
 	d.Set("self_link", disk.SelfLink)
+	d.Set("type", typeUrl[len(typeUrl)-1])
+	d.Set("zone", zoneUrl[len(zoneUrl)-1])
+	d.Set("size", disk.SizeGb)
 	if disk.DiskEncryptionKey != nil && disk.DiskEncryptionKey.Sha256 != "" {
 		d.Set("disk_encryption_key_sha256", disk.DiskEncryptionKey.Sha256)
 	}
+	if disk.SourceImage != "" {
+		imageUrl := strings.Split(disk.SourceImage, "/")
+		d.Set("image", imageUrl[len(imageUrl)-1])
+	}
+	if disk.SourceSnapshot != "" {
+		snapshotUrl := strings.Split(disk.SourceSnapshot, "/")
+		d.Set("snapshot", snapshotUrl[len(snapshotUrl)-1])
+	}
```

Comment:
It may sound like nitpick and it may be just me, but I spent good couple of minutes trying to understand what all these changes above (with `strings.Split`) mean.
So just for clarity/readability I think it would help if we renamed `zoneUrl` to `zoneUrlParts`, `typeUrl` to `typeUrlParts` etc. so that it's more obvious that these variables hold slice, not a string?